kmos_tools/star_offsets.py

```diff
@@ -117,9 +118,13 @@ def star_positions_batch(frame_list, psf_cut=0.8, edge_x=2., edge_y=2., star_ifu
 
     # Save star parameter output
     star_table = np.array(star_table)
-    med_FWHM   = np.nanmedian([float(x) for x in star_table[:, 4] if x > 0])
-    med_ba     = np.nanmedian([float(x) for x in star_table[:, 5] if x > 0])
-    med_pa     = np.nanmedian([float(x) for x in star_table[:, 6] if x > 0])
+
+    med_FWHM   = [float(x) for x in star_table[:, 4]]
```

Comment:
did the original version of this not work? I'd prefer to keep the more compact version if it does

Reply:
The original version didn't work as it was trying to check if x was > 0 before converting to a float which didn't work as x was a string at that point and I couldn't work out a more compact way of doing the conversion first and then checking if > 0.

Reply:
Hmm, ok, maybe something changed between python 2 and 3 (this was originally python 2 code). As long as it works now, then it's fine!

Could you please change these lines to the following, slightly more compact, format? 
```
FWHM  = np.array([float(x) for x in star_table[:, 4]])
med_FWHM  = np.nanmedian(FWHM[FWHM > 0.])
```
i.e. the first line makes a list, the second line get the median for values in the list > 0. And the same for `ba` and `pa`